packages/fluentui/react-northstar/src/themes/teams/components/Menu/menuItemStyles.ts

```diff
@@ -29,21 +29,29 @@ const getFocusedStyles = ({
   colors: StrictColorScheme<ItemType<typeof menuColorAreas>>;
 }): ICSSInJSStyle => {
   const { primary, underlined, active, vertical } = props;
+
   if (active && !underlined && !vertical) return {};
-  return {
-    color: v.colorActive,
-    background: v.backgroundColorFocus || colors.backgroundFocus,
 
-    ...(primary && {
-      color: colors.foregroundFocus,
+  return {
+    ...(!vertical && {
+      color: primary ? colors.foregroundFocus : v.colorActive || colors.foregroundActive,
+      background: primary ? colors.backgroundFocus : v.backgroundColorFocus || colors.backgroundFocus,
     }),
-    ...(vertical &&
-      !primary && {
-        border: `solid 1px ${v.borderColorFocus}`,
-        outline: `solid 1px ${v.outlineColorFocus}`,
-        margin: pxToRem(1),
-        background: v.verticalBackgroundColorFocus || colors.backgroundFocus,
+
+    ...(vertical && {
+      border: `solid 1px ${v.borderColorFocus}`,
+      outline: `solid 1px ${v.outlineColorFocus}`,
+      margin: pxToRem(1),
+      background: v.verticalBackgroundColorFocus,
+      color: v.colorFocus || colors.foregroundFocus,
+
+      ...(primary && { color: v.color }),
+
+      ...(active && {
+        color: primary ? colors.foregroundFocus : v.colorActive,
```

Comment:
```suggestion
        color: v.colorActive,
        ...(primary && { color: colors.foregroundFocus }),
```

Let's avoid inline ternaries for props